src/uct/ib/dc/base/dc_iface.c

```diff
@@ -134,19 +136,29 @@ static ucs_status_t uct_dc_iface_dci_connect(uct_dc_iface_t *iface, uct_rc_txqp_
 
 /* restore dc qp from error to rts state */
 ucs_status_t uct_dc_iface_dci_reconnect(uct_dc_iface_t *iface,
-                                        uct_rc_txqp_t *dci)
+                                        uct_rc_txqp_t *txqp,
+                                        ucs_class_t *ep_cls)
 {
-    struct ibv_exp_qp_attr attr;
-
-    memset(&attr, 0, sizeof(attr));
-    attr.qp_state = IBV_QPS_RESET;
+    uct_rc_mlx5_iface_t *mlx5_iface;
```

Comment:
need to avoid mlx5/verbs code in "base". maybe better to add another virtual function to rc_iface: reset_qp(iface, txqp)?